### DISCUSSION TITLE
Windows + 2*G.O. = BSOD

### DISCUSSION THREAD

**Discussion by newVideophile** [GrandOrgue/grandorgue#1365] [Help]
I have noticed that starting 2 instances of GO (of course unintentionnaly) will crash my Windows laptop. The hit rate is 100%. Last occurence a few minutes ago, when I double clicked on the GO icon.

I don't see any case were two GO instances would be useful, so may I suggest to add in future release a verification at startup that no other instance of GO is already running ? Unless the problem is not related to GO but rather to my specific PC configuration.

**Comment by midimusic** [comment]
I think the problem lies in your computer. Probably insufficient memory.

I have been running multiple copies of GO in windows for many years and never had any problems. It is a very useful feature which allows the midi output one organ to drive the midi input of another. Particularly useful when using two or more different organs at the same time.

Changing this would have no benefit, be counterproductive and reduce the usefulness of GO significantly.

csw900

**Reply by newVideophile** [reply]
Ah yes, you're right, my PC has only 8 Gb of RAM. I would have expected a clean "Out of memroy" message rather than a BSOD, though. But the fault is probably more on the Windows side.

Anyway, I'm curious to understand why you want to use two organs at the same time. To create a big one from two small ones ?

**Reply by midimusic** [reply]
GO does not cope well with insufficient memory. (This is partly caused by the operating system trying to use disk storage to cope with the shortage).

One of the benefits of running two different organs is that you can play a midi file on one organ and record midi on the other - thus translating the midi file. This is most effective when using GO alongside eplayOrgan but the principal works for most organs.

For playing from the manuals you can extend the organ you are playing to the other organ and effectively play both organs simultaneously. This will also work for more than two organs.

csw900

**Comment by larspalo** [comment]
@newVideophile Exactly how are you starting multiple instances? If done correctly, it should not crash your system.

**Reply by newVideophile** [reply]
Basically, I click twice on the GrandOrgue icon. Either because I (unintentionnaly) double click or I click once, then click again because I was thinking that my first click had not been taken into account. 

I didn't know there were "correct" and "incorrect" ways to start a program in Windows...

**Reply by larspalo** [reply]
Multiple instances of GrandOrgue should each be started from the terminal with the specification of -i instanceName after the program name so that they may be differentiated between.

![GOstartHelpDialog](https://user-images.githubusercontent.com/6173941/216693162-0c277baf-49a1-4abd-a64e-42ec95076f7a.png)

But yes, ultimately you're limited by the amount of available RAM on your system.

**Reply by midimusic** [reply]
I am not doubting that starting multiple instances of GO from a terminal would work (I have never tried it!!). However this defeats the whole point of a GUI i.e.Windows.

Multiple instances of GO may be started in Windows just by double clicking (once) a shortcut on the desktop. It is wise to allow the first instance to finish loading before double clicking again to start another instance. 

If you try to load another instance of GO too early it is asking for trouble - especially if you are short of memory. However smaller programs can be loaded multiple times without waiting.

csw900

**Reply by thbeckpr** [reply]
You can create a launcher for each instance without having to use the terminal. At least in Linux.

**Reply by larspalo** [reply]
One of the reasons of multiple instances specified with -i (and yes, you can create starters/launchers for them) is that you keep the memory spaces separate. Also another benefit is that you can have completely different settings for the different instances - even using different sound cards and settings for them. That said, it's something I'm not using on a regular basis myself so I've only very briefly tested it.

**Comment by thbeckpr** [comment]
I use it for each of my sample sets because they require different
settings.
